src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -84,15 +85,30 @@ internal ODataSerializerContext(ResourceContext resource, IEdmProperty edmProper
             {
                 SelectExpandClause = expandedNavigationSelectItem.SelectAndExpand;
             }
-
             EdmProperty = edmProperty; // should be nested property
 
+            IList<string> pathSegments = context.PathToProperty != null ? context.PathToProperty.PathSegments.ToList() : new List<string>();
+            if (edmProperty != null)
+            {
+                bool isDerived = !resource.StructuredType.AsElementType().IsOrInheritsFrom(edmProperty.DeclaringType);
+                if (isDerived)
+                {
+                    pathSegments.Add(edmProperty.DeclaringType.FullTypeName());
+                }
+                pathSegments.Add(edmProperty.Name);
```

Comment:
>pathSegments.Add(edmProperty.Name); [](start = 16, length = 35)

Rather than calling ToList<> on the path segments in order to call Add(), would it be more efficient to keep it an IEnumerable<string> and call Concat?